users/vieting/experiments/switchboard/ctc/feat/network_helpers/perturbation.py

```diff
@@ -121,17 +123,32 @@ def preemphasis_numpy(waveform, coeff=0.97):
 
     @staticmethod
     def apply_codecs(audio, sample_rate, random_state, codecs):
-        import sox
-
-        tfm = sox.Transformer()
+        output_audio = audio
         for codec in codecs:
             prob = codec.pop("prob", 1.0)
             if random_state.random() < prob:
                 if codec.get("encoding") == "ULAW":
-                    tfm.set_output_format(encoding="u-law")
+                    # standard values for µ-law encoding
+                    quantization_bits = 8
+                    MU = 255.0
+
+                    # ensure audio is normalised
+                    max_amplitude = np.max(np.abs(output_audio))
+                    normalized_audio = audio / max_amplitude
+
+                    # µ-law encoding formula
```

Comment:
```suggestion
                    # mu-law encoding formula
```
Probably fine, but sometimes special characters can break things, so I'd rather be on the safe side.